include/triton/core/tritonbackend.h

```diff
@@ -1411,6 +1411,36 @@ TRITONBACKEND_ISPEC TRITONSERVER_Error* TRITONBACKEND_GetBackendAttribute(
     TRITONBACKEND_Backend* backend,
     TRITONBACKEND_BackendAttribute* backend_attributes);
 
+/// Query the total memory usage of the model and its associated model
+/// instances. Unless TRITONBACKEND_BackendAttributeSetEnableMemoryTracker is
+/// explicitly set to true, this function is optional, a backend is not required
+/// to implement it. This function is called when changes are made to the model
+/// and Triton wants to retrieve the latest memory usage. This function may be
+/// called multiple times within the lifetime of the model (between
+/// TRITONBACKEND_ModelInitialize and TRITONBACKEND_ModelFinalize).
+/// The backend may return error to indicate failure to get the latest memory
+/// usage.
+///
+/// \param model The model.
+/// \param usage The list of buffer attributes that the backend can record the
+/// memory usage, each entry should record the total memory usage of a given
+/// memory type and id. For example, if the model itself occupies 64 bytes on
+/// CUDA device 0 and it has two model instances that each instance occupies
+/// 128 bytes on CUDA device 1. Then 'usage' should have first two entries set,
+/// one has the buffer attributes of "type GPU, id 0, 64 bytes" and the
+/// other has "type GPU, id 1, 256 bytes". The list and the list entries are
+/// owned by Triton.
+/// \param usage_size This parameter is both input and output of the function.
```

Comment:
Why not delegate this responsibility completely to the backend and make this only an output parameter? For example, it can create any number of buffer attributes it needs for reporting memory usage.